crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -595,3 +596,685 @@ async fn cow_amm_driver_support(web3: Web3) {
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(bob.private_key()).unwrap()),
+    );
+    let user_order_id = services.create_order(&user_order).await.unwrap();
+
+    // Set the fees appropriately
+    let fee_cow_amm = U256::exp10(16); // 0.01 WETH
+    let fee_user = to_wei(1); // 1 DAI
+
+    // Configure the mock solver's solution
+    let mocked_solution = Solution {
+        id: 1,
+        prices: HashMap::from([
+            (dai.address(), to_wei(1)),                         // 1 DAI = $1
+            (onchain.contracts().weth.address(), to_wei(2000)), // 1 WETH = $2000
+        ]),
+        trades: vec![
+            solvers_dto::solution::Trade::Jit(solvers_dto::solution::JitTrade {
+                order: solvers_dto::solution::JitOrder {
+                    sell_token: cow_amm_order.sell_token,
+                    buy_token: cow_amm_order.buy_token,
+                    receiver: cow_amm_order.receiver.unwrap_or_default(),
+                    sell_amount: cow_amm_order.sell_amount,
+                    buy_amount: cow_amm_order.buy_amount,
+                    valid_to: cow_amm_order.valid_to,
+                    app_data: cow_amm_order.app_data.0,
+                    kind: Kind::Sell,
+                    sell_token_balance: SellTokenBalance::Erc20,
+                    buy_token_balance: BuyTokenBalance::Erc20,
+                    signing_scheme: SigningScheme::Eip1271,
+                    signature,
+                },
+                executed_amount: cow_amm_order.sell_amount - fee_cow_amm,
+                fee: Some(fee_cow_amm),
+            }),
+            solvers_dto::solution::Trade::Fulfillment(solvers_dto::solution::Fulfillment {
+                order: user_order_id.0,
+                executed_amount: user_order.sell_amount - fee_user,
+                fee: Some(fee_user),
+            }),
+        ],
+        pre_interactions: vec![cow_amm_commitment],
+        interactions: vec![],
+        post_interactions: vec![],
+        gas: None,
+    };
+    mock_solver.configure_solution(Some(mocked_solution.clone()));
+
+    // Drive solution
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async {
+        let amm_weth_balance_after = onchain
+            .contracts()
+            .weth
+            .balance_of(cow_amm.address())
+            .call()
+            .await
+            .unwrap();
+        let bob_weth_balance_after = onchain
+            .contracts()
+            .weth
+            .balance_of(bob.address())
+            .call()
+            .await
+            .unwrap();
+
+        let amm_weth_sent = amm_weth_balance_before - amm_weth_balance_after;
+        let bob_weth_received = bob_weth_balance_after - bob_weth_balance_before;
+
+        // Bob should receive WETH, CoW AMM's WETH balance decreases
+        bob_weth_received >= user_order.buy_amount && amm_weth_sent >= cow_amm_order.sell_amount
```

Comment:
If you configure a sell order you should never have more tokens leave your account than you signed.
```suggestion
        bob_weth_received >= user_order.buy_amount && amm_weth_sent == cow_amm_order.sell_amount
```